test/e2e/remote_dns_name/main_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the License);
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an AS IS BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integ
+// +build integ
+
+package remote_dns_name
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+
+	"github.com/openshift-service-mesh/federation/test/e2e/common"
+	"github.com/openshift-service-mesh/federation/test/e2e/common/coredns"
+)
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite(m).
+		RequireMinClusters(2).
+		RequireMaxClusters(2).
+		Setup(common.RecreateControlPlaneNamespace).
+		Setup(common.CreateCACertsSecret).
+		Setup(common.DeployControlPlanes("k8s")).
+		Setup(coredns.PatchHosts).
+		Setup(common.InstallOrUpgradeFederationControllers(common.InstallOptions{SetRemoteDNSName: true})).
+		Setup(namespace.Setup(&common.AppNs, namespace.Config{Prefix: "app", Inject: true})).
+		// a - client
+		// b - service available in east and west clusters - covers importing with WorkloadEntry
+		// c - service available only in west cluster - covers importing with ServiceEntry
+		Setup(common.DeployApps(&common.EastApps, common.EastClusterName, namespace.Future(&common.AppNs), false, "a", "b")).
+		Setup(common.DeployApps(&common.WestApps, common.WestClusterName, namespace.Future(&common.AppNs), false, "b", "c")).
+		// c must be removed from the east cluster, because we want to test importing a service
+		// that exists only in the remote cluster.
+		Setup(common.RemoveServiceFromClusters("c", namespace.Future(&common.AppNs), common.EastClusterName)).
```

Comment:
nit: there's no need to remove "c" app from East cluster, you don't deploy it there, do you?

Reply:
This function creates Deployments only in the west cluster, but Services are created in all clusters.
```
Setup(common.DeployApps(&common.WestApps, common.WestClusterName, namespace.Future(&common.AppNs), false, "b", "c")).
```
Istio test framework was designed to support pure multi-cluster scenarios, so I guess this is not a bug, but a feature.

Reply:
I need to dig deeper to better understand how it works under the hood. For the novice like me it reads a bit counter intuitive.